modules/Kruskal_algorithm/src/Kruskal_algorithm.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright 2021 Volkova Anastasia
+
+#include <utility>
+#include <algorithm>
+#include <vector>
+#include <random>
+#include <ctime>
+#include <numeric>
+
+#include "include/Kruskal_algorithm.h"
+
+Graph::Graph(const int size) {
+    n = size;
+    parent.resize(n, 0);
+    std::iota(parent.begin(), parent.end(), 0);
+}
+
+Graph::Graph(const Graph& tmp) : n(tmp.n),
+                                 G(tmp.G),
+                                 MST(tmp.MST),
+                                 parent(tmp.parent) {}
+
+Graph& Graph::operator=(const Graph& tmp) {
+    if (this == &tmp) return *this;
+    n = tmp.get_n();
+    G = tmp.get_G();
+    MST = tmp.get_MST();
+    parent = tmp.get_parent();
+    return *this;
+}
+
+void Graph::addEdge(const int x, const int y, const int w) {
+    G.emplace_back(make_pair(w, edge(x, y)));
+}
+
+int Graph::find_set(int vertex) {
+    if (vertex == parent[vertex]) return vertex;
+    return parent[vertex] = find_set(parent[vertex]);
```

Comment:
Лучше не использовать подобные конструкции без видимой причины, а писать отдельными выражениями:
```suggestion
    parent[vertex] = find_set(parent[vertex]);
    return parent[vertex];
```